### PR TITLE
[CDAP-11977] DataPrep add Columns Replace directive

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-11977

### CODE REVIEW

File: cdap-ui/app/cdap/components/DataPrep/Directives/ColumnActions/ReplaceColumns/index.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        <ModalFooter>
+          <fieldset disabled={this.state.loading}>
+            <button
+              className="btn btn-primary"
+              onClick={this.applyDirective}
+              disabled={!isNil(this.state.error) || isEmpty(this.state.sourcePattern)}
+            >
+              {
+                this.state.loading ?
+                  <span className="fa fa-spin fa-spinner"></span>
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/Directives/ColumnActions/ReplaceColumns/index.js]: Use self-closing tag?

**Review by tonybach — COMMENTED** [review]
One minor comment, otherwise code LGTM 👍

**Comment by tonybach** [comment]
Also, do we need to add autofocus on one of the input fields?

**Comment by ajainarayanan** [comment]
Code LGTM 👍

**Comment by bdmogal** [comment]
image

The options should be:
1. Remove Prefix - Accepts only prefix. Placeholder: ``e.g. body_, work_``
2. Remove Pattern - This was "Remove Suffix" in the spec, but let's change it to Remove any pattern. - Accepts the pattern to replace only. Placeholder: ``e.g. eol$,  body, ``
3. Replace Custom Pattern:
- Accepts a custom pattern to replace, and a pattern to replace with.

**Comment by bdmogal** [comment]
@tonybach 
image

1. Alignment changes - align (left) all the textboxes to the text of the radio button options. Also align "Replace With" similarly? 
2. Please change the placeholder for "Remove pattern" with ``e.g. body_[0-9]{2}, eol$``
3. Change the text "Custom Pattern" to "Replace custom pattern"
4. Change "else add a replacement text" to "else add a replacement pattern"
5. The custom pattern should be as: 
```
Replace
<Text box for pattern to be replaced>
With
<Textbox for pattern to replace with>
```

**Review by bdmogal — APPROVED** [review]
@tonybach 

1. "Replace Custom Pattern" - > "Replace Pattern"
2. Align Replace and With with the textboxes.

Other than that, LGTM

**Pull request merged**